Extensions/Video/JsExtension.js

```diff
@@ -0,0 +1,775 @@
+        objectContent.property2 = newValue === "1";
+        return true;
+      }
+      if (propertyName === "Volume") {
+        objectContent.property3 = newValue;
+        return true;
+      }
+
+      return false;
+    };
+    videoObject.getProperties = function(objectContent) {
+      var objectProperties = new gd.MapStringPropertyDescriptor();
+
+      objectProperties.set(
+        "Video object opacity",
+        new gd.PropertyDescriptor(objectContent.property1)
+      );
+      objectProperties.set(
+        "Video Looped or no",
+        new gd.PropertyDescriptor(
+          objectContent.property2 ? "true" : "false"
+        ).setType("boolean")
+      );
+      objectProperties.set(
+        "Volume",
+        new gd.PropertyDescriptor(objectContent.property3.toString()).setType(
+          "number"
+        )
+      );
+
+      return objectProperties;
+    };
+    videoObject.setRawJSONContent(
+      JSON.stringify({
+        property1: "Hello world",
+        property2: false,
+        property3: 100
+      })
+    );
+
+    videoObject.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      if (propertyName === "Video instance opacity") {
+        instance.setRawStringProperty("instanceprop1", newValue);
+        return true;
+      }
+      if (propertyName === "My other instance property") {
+        instance.setRawFloatProperty("instanceprop2", parseFloat(newValue));
+        return true;
+      }
+
+      return false;
+    };
+    videoObject.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+
+      instanceProperties.set(
+        "Video instance opacity",
+        new gd.PropertyDescriptor(
+          instance.getRawStringProperty("instanceprop1")
+        )
+      );
+      instanceProperties.set(
+        "My other instance property",
+        new gd.PropertyDescriptor(
+          instance.getRawFloatProperty("instanceprop2").toString()
+        ).setType("number")
+      );
+
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        "VideoObject",
+        _("Video object for testing"),
+        _("This video object does nothing"),
+        "JsPlatform/Extensions/videoicon32.png",
+        videoObject
+      )
+      .setIncludeFile("Extensions/Video/videoruntimeobject.js")
+      .addIncludeFile("Extensions/Video/videoruntimeobject-pixi-renderer.js");
+
+    object
+      .addAction(
+        "Play",
+        _("Play an video"),
+        _("Play an video"),
+        _("Play the video of : _PARAM0_"),
+        "",
+        "JsPlatform/Extensions/videoicon24.png",
+        "JsPlatform/Extensions/videoicon16.png"
+      )
+      .addParameter("object", _("Choose a video object"), "", false)
```

Comment:
"Choose a video object" => "Video object". Also, the third argument must be the type of the object:

```
.addParameter("object", _("Video object"),  "VideoObject", false)
```